<a href="https://colab.research.google.com/github/Ghefira233/Data-Project/blob/main/Metode_Numerik_Menghitung_Turunan_Numerik.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#Menghitung f'(2.4)
print("Halo, selamat datang. Di program ini, kamu dapat menghitung turunan numerik menggunakan 4 metode")    
p = input("Mau tau ada apa aja? (Y/N) : ")                                                                   #Meminta untuk memilih ingin menampilkan 4 metode atau tidak

if p == "Y":                                                                                                 #Jika memilih Y, maka akan meng-print "Oke, ada TPEP, TPMP, FPEP, FPMP. Silahkan mencoba :D"
  print("Oke, ada TPEP, TPMP, FPEP, FPMP. Silahkan mencoba :D")
elif p == "N":                                                                                               #Jika memilih N, maka akan meng-print "Oke udah hafal berarti, langsung lanjut isi fungsinya!"
  print("Oke udah hafal berarti, langsung lanjut isi fungsinya!")

pilih = int(input('Mau pake metode apa? (1 = TPEP, 2 = TPMP, 3 = FPEP, 4 = FPMP) : '))                       #Meminta untuk memilih ingin pakai metode apa
if pilih == 1:                                                                                               #Jika pilih 1, maka akan menjalankan program untuk TPEP 
  x = [2.4,2.5,2.6,2.7,2.8,2.9,3.0]                                                                          #Titik-titik x berupa list
  y = [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859]                                              #Titik-titik y berupa list
  x_turun = eval(input('Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : '))               #Meminta untuk menginput titik dimana nilai turunan fungsinya akan diaproksimasi
  h = eval(input('Masukkan stepsize nya : '))                                                                #Meminta untuk menginput stepsize nya
  n = int(input('Ingin diturunkan berapa kali? : '))                                                         #Meminta untuk memilih ingin diturunkan berapa kali                                                                                              
  
  def TPEP(x0,h,x,y,n):
    if n==1:                                                                                                 #Jika n = 1, maka fungsi akan diturunkan sebanyak 1 kali
      return (1/(2*h))*(-3*y[x.index(round(x0,1))]+4*y[x.index(round(x0+h,1))]-y[x.index(round(x0+2*h,1))])
    else:                                                                                                    #Untuk n lainnya maka fungsi akan diturunkan sebanyak n kali
      return (1/(2*h))*(-3*TPEP(x0,h,x,y,n-1)+4*TPEP(x0+h,h,x,y,n-1)-TPEP(x0+2*h,h,x,y,n-1))

  print(TPEP(x_turun,h,x,y,n))                                                                               #Meng-print hasil turunannya

if pilih == 2:                                                                                               #Jika pilih 2, maka akan menjalankan program untuk TPMP
  x = [2.4,2.5,2.6,2.7,2.8,2.9,3.0]                                                                          #Titik-titik x berupa list
  y = [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859]                                              #Titik-titik y berupa list
  x_turun2 = eval(input('Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : '))              #Meminta untuk menginput titik dimana nilai turunan fungsinya akan diaproksimasi
  h = eval(input('Masukkan stepsize nya : '))                                                                #Meminta untuk menginput stepsize nya
  n = int(input('Ingin diturunkan berapa kali? : '))                                                         #Meminta untuk memilih ingin diturunkan berapa kali                                                                                                                       

  def TPMP(x0,h,x,y,n):     
    if n==1:                                                                                                 #Jika n = 1, maka fungsi akan diturunkan sebanyak 1 kali
      return (1/(2*h))*((y[x.index(round(x0+h,1))])-(y[x.index(round(x0-h,1))]))
    else:                                                                                                    #Untuk n lainnya maka fungsi akan diturunkan sebanyak n kali
      return (1/(2*h))*(TPMP(x0+h,h,x,y,n-1)-TPMP(x0-h,h,x,y,n-1))                             

  print(TPMP(x_turun2,h,x,y,n))                                                                              #Meng-print hasil turunannya

if pilih == 3:                                                                                               #Jika pilih 3, maka akan menjalankan program untuk TPMP
  x = [2.4,2.5,2.6,2.7,2.8,2.9,3.0]                                                                          #Titik-titik x berupa list
  y = [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859]                                              #Titik-titik y berupa list
  x_turun3 = eval(input('Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : '))              #Meminta untuk menginput titik dimana nilai turunan fungsinya akan diaproksimasi
  h = eval(input('Masukkan stepsize nya : '))                                                                #Meminta untuk menginput stepsize nya
  n = int(input('Ingin diturunkan berapa kali? : '))                                                         #Meminta untuk memilih ingin diturunkan berapa kali                                                                                                                       

  def FPEP(x0,h,x,y,n):
    if n==1:                                                                                                 #Jika n = 1, maka fungsi akan diturunkan sebanyak 1 kali
      return (1/(12*h))*(-25*(y[x.index(round(x0,1))])+48*(y[x.index(round(x0+h,1))])-36*(y[x.index(round(x0+2*h,1))])+16*(y[x.index(round(x0+3*h,1))])-3*(y[x.index(round(x0+4*h,1))]))
    else:                                                                                                    #Untuk n lainnya maka fungsi akan diturunkan sebanyak n kali
      return (1/(12*h))*(-25*FPEP(x0,h,x,y,n-1)+48*FPEP(x0+h,h,x,y,n-1)-36*FPEP(x0+2*h,h,x,y,n-1)+16*FPEP(x0+3*h,h,x,y,n-1)-3*FPEP(x0+4*h,h,x,y,n-1))                                                                           
  
  print(FPEP(x_turun3,h,x,y,n))                                                                              #Meng-print hasil turunannya

else:                                                                                                        #Jika pilih 4, maka akan menjalankan program untuk TPMP 
  x = [2.4,2.5,2.6,2.7,2.8,2.9,3.0]                                                                          #Titik-titik x berupa list
  y = [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859]                                              #Titik-titik y berupa list
  x_turun4 = eval(input('Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : '))              #Meminta untuk menginput titik dimana nilai turunan fungsinya akan diaproksimasi
  h = eval(input('Masukkan stepsize nya : '))                                                                #Meminta untuk menginput stepsize nya
  n = int(input('Ingin diturunkan berapa kali? : '))                                                         #Meminta untuk memilih ingin diturunkan berapa kali                                                                                                                    
  
  def FPMP(titik_x,titik_y,x0,h,n):
    if n==1:                                                                                                 #Jika n = 1, maka fungsi akan diturunkan sebanyak 1 kali
      return (1/(12*h))*((y[x.index(round(x0-2*h,1))])-8*(y[x.index(round(x0-h,1))])+8*f(y[x.index(round(x0+h,1))])-(y[x.index(round(x0-2*h,1))]))
    else:                                                                                                    #Untuk n lainnya maka fungsi akan diturunkan sebanyak n kali
      return (1/(12*h))*(FPMP(x0-2*h,h,x,y,n-1)-8*FPMP(x0-h,h,x,y,n-1)+8*FPMP(x0+h,h,x,y,n-1)-FPMP(x0-2*h,h,x,y,n-1))                        

  print(FPMP(x_turun4,h,x,y,n))                                                                              #Meng-print hasil turunannya

Halo, selamat datang. Di program ini, kamu dapat menghitung turunan numerik menggunakan 4 metode
Mau tau ada apa aja? (Y/N) : Y
Oke, ada TPEP, TPMP, FPEP, FPMP. Silahkan mencoba :D
Mau pake metode apa? (1 = TPEP, 2 = TPMP, 3 = FPEP, 4 = FPMP) : 3
Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : 2.4
Masukkan stepsize nya : 0.1
Ingin diturunkan berapa kali? : 1
1.1604999999999999


Program ini berfungsi untuk menghitung turunan numerik menggunakan 4 metode.

*Input yang diharapkan:*
1. Titik $x$ dan nilai dari $f(x)$
2. Pilihan metode yang ingin digunakan

*Output yang diharapkan:*
1. Nilai dari turunan $f(x)$ di titik tersebut

*Algoritmanya:*
1. Program ini akan meminta untuk memilih ingin menggunakan metode apa. **1** untuk TPEP, **2** untuk  TPMP, **3** untuk FPEP, **4** untuk FPMP
2. **Apabila memilih 1, maka akan menjalankan program untuk menghitung turunan pada titik x_turun dengan TPEP**
3. Memasukkan titik-titik $x$ [2.4,2.5,2.6,2.7,2.8,2.9,3.0] ke dalam variabel **x** dan titik-titik $y$ [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859] ke dalam variabel **y**
4. Selanjutnya diminta untuk meng-input titik dimana nilai turunan fungsinya akan di aproksimasi yang dimasukkan ke dalam variabel **x_turun**, stepsizenya yang dimasukkan ke dalam variabel **h** dan ingin diturunkan berapa kali ke dalam variabel **n**
5. Meng-print hasil turunannya pada titik **x_turun** dengan metode TPEP
6. **Apabila memilih 2, maka akan menjalankan program untuk menghitung turunan pada nilai $x_0$ dengan TPMP**
7. Memasukkan titik-titik $x$ [2.4,2.5,2.6,2.7,2.8,2.9,3.0] ke dalam variabel **x** dan titik-titik $y$ [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859] ke dalam variabel **y**
8. Selanjutnya diminta untuk meng-input titik dimana nilai turunan fungsinya akan di aproksimasi yang dimasukkan ke dalam variabel **x_turun**, stepsizenya yang dimasukkan ke dalam variabel **h** dan ingin diturunkan berapa kali ke dalam variabel **n**
9. Meng-print hasil turunannya pada titik **x_turun** dengan metode TPMP
10. **Apabila memilih 3, maka akan menjalankan program untuk menghitung turunan pada nilai $x_0$ dengan FPEP**
11. Memasukkan titik-titik $x$ [2.4,2.5,2.6,2.7,2.8,2.9,3.0] ke dalam variabel **x** dan titik-titik $y$ [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859] ke dalam variabel **y**
12. Selanjutnya diminta untuk meng-input titik dimana nilai turunan fungsinya akan di aproksimasi yang dimasukkan ke dalam variabel **x_turun**, stepsizenya yang dimasukkan ke dalam variabel **h** dan ingin diturunkan berapa kali ke dalam variabel **n**
13. Meng-print hasil turunannya pada titik **x_turun** dengan metode FPEP
14. **Apabila memilih 4, maka akan menjalankan program untuk menghitung turunan pada nilai $x_0$ dengan FPMP**
15. Memasukkan titik-titik $x$ [2.4,2.5,2.6,2.7,2.8,2.9,3.0] ke dalam variabel **x** dan titik-titik $y$ [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859] ke dalam variabel **y**
16. Selanjutnya diminta untuk meng-input titik dimana nilai turunan fungsinya akan di aproksimasi yang dimasukkan ke dalam variabel **x_turun**, stepsizenya yang dimasukkan ke dalam variabel **h** dan ingin diturunkan berapa kali ke dalam variabel **n**
17. Meng-print hasil turunannya pada titik **x_turun** dengan metode FPMP

In [ ]:
#Menghitung f''(2.5)
print("Halo, selamat datang. Di program ini, kamu dapat menghitung turunan numerik menggunakan 4 metode")    
p = input("Mau tau ada apa aja? (Y/N) : ")                                                                   #Meminta untuk memilih ingin menampilkan 4 metode atau tidak

if p == "Y":                                                                                                 #Jika memilih Y, maka akan meng-print "Oke, ada TPEP, TPMP, FPEP, FPMP. Silahkan mencoba :D"
  print("Oke, ada TPEP, TPMP, FPEP, FPMP. Silahkan mencoba :D")
elif p == "N":                                                                                               #Jika memilih N, maka akan meng-print "Oke udah hafal berarti, langsung lanjut isi fungsinya!"
  print("Oke udah hafal berarti, langsung lanjut isi fungsinya!")

pilih = int(input('Mau pake metode apa? (1 = TPEP, 2 = TPMP, 3 = FPEP, 4 = FPMP) : '))                       #Meminta untuk memilih ingin pakai metode apa
if pilih == 1:                                                                                               #Jika pilih 1, maka akan menjalankan program untuk TPEP 
  x = [2.4,2.5,2.6,2.7,2.8,2.9,3.0]                                                                          #Titik-titik x berupa list
  y = [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859]                                              #Titik-titik y berupa list
  x_turun = eval(input('Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : '))               #Meminta untuk menginput titik dimana nilai turunan fungsinya akan diaproksimasi
  h = eval(input('Masukkan stepsize nya : '))                                                                #Meminta untuk menginput stepsize nya
  n = int(input('Ingin diturunkan berapa kali? : '))                                                         #Meminta untuk memilih ingin diturunkan berapa kali                                                                                              
  
  def TPEP(x0,h,x,y,n):
    if n==1:                                                                                                 #Jika n = 1, maka fungsi akan diturunkan sebanyak 1 kali
      return (1/(2*h))*(-3*y[x.index(round(x0,1))]+4*y[x.index(round(x0+h,1))]-y[x.index(round(x0+2*h,1))])
    else:                                                                                                    #Untuk n lainnya maka fungsi akan diturunkan sebanyak n kali
      return (1/(2*h))*(-3*TPEP(x0,h,x,y,n-1)+4*TPEP(x0+h,h,x,y,n-1)-TPEP(x0+2*h,h,x,y,n-1))

  print(TPEP(x_turun,h,x,y,n))                                                                               #Meng-print hasil turunannya

if pilih == 2:                                                                                               #Jika pilih 2, maka akan menjalankan program untuk TPMP
  x = [2.4,2.5,2.6,2.7,2.8,2.9,3.0]                                                                          #Titik-titik x berupa list
  y = [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859]                                              #Titik-titik y berupa list
  x_turun2 = eval(input('Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : '))              #Meminta untuk menginput titik dimana nilai turunan fungsinya akan diaproksimasi
  h = eval(input('Masukkan stepsize nya : '))                                                                #Meminta untuk menginput stepsize nya
  n = int(input('Ingin diturunkan berapa kali? : '))                                                         #Meminta untuk memilih ingin diturunkan berapa kali                                                                                                                       

  def TPMP(x0,h,x,y,n):     
    if n==1:                                                                                                 #Jika n = 1, maka fungsi akan diturunkan sebanyak 1 kali
      return (1/(2*h))*((y[x.index(round(x0+h,1))])-(y[x.index(round(x0-h,1))]))
    else:                                                                                                    #Untuk n lainnya maka fungsi akan diturunkan sebanyak n kali
      return (1/(2*h))*(TPMP(x0+h,h,x,y,n-1)-TPMP(x0-h,h,x,y,n-1))                             

  print(TPMP(x_turun2,h,x,y,n))                                                                              #Meng-print hasil turunannya

if pilih == 3:                                                                                               #Jika pilih 3, maka akan menjalankan program untuk TPMP
  x = [2.4,2.5,2.6,2.7,2.8,2.9,3.0]                                                                          #Titik-titik x berupa list
  y = [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859]                                              #Titik-titik y berupa list
  x_turun3 = eval(input('Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : '))              #Meminta untuk menginput titik dimana nilai turunan fungsinya akan diaproksimasi
  h = eval(input('Masukkan stepsize nya : '))                                                                #Meminta untuk menginput stepsize nya
  n = int(input('Ingin diturunkan berapa kali? : '))                                                         #Meminta untuk memilih ingin diturunkan berapa kali                                                                                                                       

  def FPEP(x0,h,x,y,n):
    if n==1:                                                                                                 #Jika n = 1, maka fungsi akan diturunkan sebanyak 1 kali
      return (1/(12*h))*(-25*(y[x.index(round(x0,1))])+48*(y[x.index(round(x0+h,1))])-36*(y[x.index(round(x0+2*h,1))])+16*(y[x.index(round(x0+3*h,1))])-3*(y[x.index(round(x0+4*h,1))]))
    else:                                                                                                    #Untuk n lainnya maka fungsi akan diturunkan sebanyak n kali
      return (1/(12*h))*(-25*FPEP(x0,h,x,y,n-1)+48*FPEP(x0+h,h,x,y,n-1)-36*FPEP(x0+2*h,h,x,y,n-1)+16*FPEP(x0+3*h,h,x,y,n-1)-3*FPEP(x0+4*h,h,x,y,n-1))                                                                           
  
  print(FPEP(x_turun3,h,x,y,n))                                                                              #Meng-print hasil turunannya

else:                                                                                                        #Jika pilih 4, maka akan menjalankan program untuk TPMP 
  x = [2.4,2.5,2.6,2.7,2.8,2.9,3.0]                                                                          #Titik-titik x berupa list
  y = [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859]                                              #Titik-titik y berupa list
  x_turun4 = eval(input('Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : '))              #Meminta untuk menginput titik dimana nilai turunan fungsinya akan diaproksimasi
  h = eval(input('Masukkan stepsize nya : '))                                                                #Meminta untuk menginput stepsize nya
  n = int(input('Ingin diturunkan berapa kali? : '))                                                         #Meminta untuk memilih ingin diturunkan berapa kali                                                                                                                    
  
  def FPMP(x0,h,x,y,n):
    if n==1:                                                                                                 #Jika n = 1, maka fungsi akan diturunkan sebanyak 1 kali
      return (1/(12*h))*((y[x.index(round(x0-2*h,1))])-8*(y[x.index(round(x0-h,1))])+8*f(y[x.index(round(x0+h,1))])-(y[x.index(round(x0-2*h,1))]))
    else:                                                                                                    #Untuk n lainnya maka fungsi akan diturunkan sebanyak n kali
      return (1/(12*h))*(FPMP(x0-2*h,h,x,y,n-1)-8*FPMP(x0-h,h,x,y,n-1)+8*FPMP(x0+h,h,x,y,n-1)-FPMP(x0-2*h,h,x,y,n-1))                        

  print(FPMP(x_turun4,h,x,y,n))                                                                              #Meng-print hasil turunannya

Halo, selamat datang. Di program ini, kamu dapat menghitung turunan numerik menggunakan 4 metode
Mau tau ada apa aja? (Y/N) : Y
Oke, ada TPEP, TPMP, FPEP, FPMP. Silahkan mencoba :D
Mau pake metode apa? (1 = TPEP, 2 = TPMP, 3 = FPEP, 4 = FPMP) : 1
Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : 2.5
Masukkan stepsize nya : 0.1
Ingin diturunkan berapa kali? : 2
-1.041499999999984
Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : 


SyntaxError: ignored

In [ ]:
#Menghitung f'''(2.7)
print("Halo, selamat datang. Di program ini, kamu dapat menghitung turunan numerik menggunakan 4 metode")    
p = input("Mau tau ada apa aja? (Y/N) : ")                                                                   #Meminta untuk memilih ingin menampilkan 4 metode atau tidak

if p == "Y":                                                                                                 #Jika memilih Y, maka akan meng-print "Oke, ada TPEP, TPMP, FPEP, FPMP. Silahkan mencoba :D"
  print("Oke, ada TPEP, TPMP, FPEP, FPMP. Silahkan mencoba :D")
elif p == "N":                                                                                               #Jika memilih N, maka akan meng-print "Oke udah hafal berarti, langsung lanjut isi fungsinya!"
  print("Oke udah hafal berarti, langsung lanjut isi fungsinya!")

pilih = int(input('Mau pake metode apa? (1 = TPEP, 2 = TPMP, 3 = FPEP, 4 = FPMP) : '))                       #Meminta untuk memilih ingin pakai metode apa
if pilih == 1:                                                                                               #Jika pilih 1, maka akan menjalankan program untuk TPEP 
  x = [2.4,2.5,2.6,2.7,2.8,2.9,3.0]                                                                          #Titik-titik x berupa list
  y = [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859]                                              #Titik-titik y berupa list
  x_turun = eval(input('Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : '))               #Meminta untuk menginput titik dimana nilai turunan fungsinya akan diaproksimasi
  h = eval(input('Masukkan stepsize nya : '))                                                                #Meminta untuk menginput stepsize nya
  n = int(input('Ingin diturunkan berapa kali? : '))                                                         #Meminta untuk memilih ingin diturunkan berapa kali                                                                                              
  
  def TPEP(x0,h,x,y,n):
    if n==1:                                                                                                 #Jika n = 1, maka fungsi akan diturunkan sebanyak 1 kali
      return (1/(2*h))*(-3*y[x.index(round(x0,1))]+4*y[x.index(round(x0+h,1))]-y[x.index(round(x0+2*h,1))])
    else:                                                                                                    #Untuk n lainnya maka fungsi akan diturunkan sebanyak n kali
      return (1/(2*h))*(-3*TPEP(x0,h,x,y,n-1)+4*TPEP(x0+h,h,x,y,n-1)-TPEP(x0+2*h,h,x,y,n-1))

  print(TPEP(x_turun,h,x,y,n))                                                                               #Meng-print hasil turunannya

if pilih == 2:                                                                                               #Jika pilih 2, maka akan menjalankan program untuk TPMP
  x = [2.4,2.5,2.6,2.7,2.8,2.9,3.0]                                                                          #Titik-titik x berupa list
  y = [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859]                                              #Titik-titik y berupa list
  x_turun2 = eval(input('Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : '))              #Meminta untuk menginput titik dimana nilai turunan fungsinya akan diaproksimasi
  h = eval(input('Masukkan stepsize nya : '))                                                                #Meminta untuk menginput stepsize nya
  n = int(input('Ingin diturunkan berapa kali? : '))                                                         #Meminta untuk memilih ingin diturunkan berapa kali                                                                                                                       

  def TPMP(x0,h,x,y,n):     
    if n==1:                                                                                                 #Jika n = 1, maka fungsi akan diturunkan sebanyak 1 kali
      return (1/(2*h))*((y[x.index(round(x0+h,1))])-(y[x.index(round(x0-h,1))]))
    else:                                                                                                    #Untuk n lainnya maka fungsi akan diturunkan sebanyak n kali
      return (1/(2*h))*(TPMP(x0+h,h,x,y,n-1)-TPMP(x0-h,h,x,y,n-1))                             

  print(TPMP(x_turun2,h,x,y,n))                                                                              #Meng-print hasil turunannya

if pilih == 3:                                                                                               #Jika pilih 3, maka akan menjalankan program untuk TPMP
  x = [2.4,2.5,2.6,2.7,2.8,2.9,3.0]                                                                          #Titik-titik x berupa list
  y = [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859]                                              #Titik-titik y berupa list
  x_turun3 = eval(input('Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : '))              #Meminta untuk menginput titik dimana nilai turunan fungsinya akan diaproksimasi
  h = eval(input('Masukkan stepsize nya : '))                                                                #Meminta untuk menginput stepsize nya
  n = int(input('Ingin diturunkan berapa kali? : '))                                                         #Meminta untuk memilih ingin diturunkan berapa kali                                                                                                                       

  def FPEP(x0,h,x,y,n):
    if n==1:                                                                                                 #Jika n = 1, maka fungsi akan diturunkan sebanyak 1 kali
      return (1/(12*h))*(-25*(y[x.index(round(x0,1))])+48*(y[x.index(round(x0+h,1))])-36*(y[x.index(round(x0+2*h,1))])+16*(y[x.index(round(x0+3*h,1))])-3*(y[x.index(round(x0+4*h,1))]))
    else:                                                                                                    #Untuk n lainnya maka fungsi akan diturunkan sebanyak n kali
      return (1/(12*h))*(-25*FPEP(x0,h,x,y,n-1)+48*FPEP(x0+h,h,x,y,n-1)-36*FPEP(x0+2*h,h,x,y,n-1)+16*FPEP(x0+3*h,h,x,y,n-1)-3*FPEP(x0+4*h,h,x,y,n-1))                                                                           
  
  print(FPEP(x_turun3,h,x,y,n))                                                                              #Meng-print hasil turunannya

else:                                                                                                        #Jika pilih 4, maka akan menjalankan program untuk TPMP 
  x = [2.4,2.5,2.6,2.7,2.8,2.9,3.0]                                                                          #Titik-titik x berupa list
  y = [1.76601,1.87697,1.97803,2.06988,2.15357,2.22953,2.29859]                                              #Titik-titik y berupa list
  x_turun4 = eval(input('Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : '))              #Meminta untuk menginput titik dimana nilai turunan fungsinya akan diaproksimasi
  h = eval(input('Masukkan stepsize nya : '))                                                                #Meminta untuk menginput stepsize nya
  n = int(input('Ingin diturunkan berapa kali? : '))                                                         #Meminta untuk memilih ingin diturunkan berapa kali                                                                                                                    
  
  def FPMP(x0,h,x,y,n):
    if n==1:                                                                                                 #Jika n = 1, maka fungsi akan diturunkan sebanyak 1 kali
      return (1/(12*h))*((y[x.index(round(x0-2*h,1))])-8*(y[x.index(round(x0-h,1))])+8*f(y[x.index(round(x0+h,1))])-(y[x.index(round(x0-2*h,1))]))
    else:                                                                                                    #Untuk n lainnya maka fungsi akan diturunkan sebanyak n kali
      return (1/(12*h))*(FPMP(x0-2*h,h,x,y,n-1)-8*FPMP(x0-h,h,x,y,n-1)+8*FPMP(x0+h,h,x,y,n-1)-FPMP(x0-2*h,h,x,y,n-1))                        

  print(FPMP(x_turun4,h,x,y,n))                                                                              #Meng-print hasil turunannya

Halo, selamat datang. Di program ini, kamu dapat menghitung turunan numerik menggunakan 4 metode
Mau tau ada apa aja? (Y/N) : N
Oke udah hafal berarti, langsung lanjut isi fungsinya!
Mau pake metode apa? (1 = TPEP, 2 = TPMP, 3 = FPEP, 4 = FPMP) : 2
Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : 2.7
Masukkan stepsize nya : 0.1
Ingin diturunkan berapa kali? : 3
0.7449999999998846
Masukkan titik dimana nilai turunan fungsinya akan diaproksimasi : 


SyntaxError: ignored

Jadi, didapat
- f'(2.4) = 1.1604999999999999 menggunakan metode FPEP
- f''(2.5) = -1.041499999999984 menggunakan metode TPEP
- f'''(2.7) = 0.7449999999998846 menggunakan metode TPMP
